src/backends/torch/torchlib.cc

```diff
@@ -534,15 +561,30 @@ namespace dd
     try
       {
         inputc.transform(ad);
-        if (_module._native)
+
+        if (_template.find("nbeats") != std::string::npos)
+          {
+            CSVTSTorchInputFileConn *ctip
+                = reinterpret_cast<CSVTSTorchInputFileConn *>(&inputc);
+            _module._native = std::make_shared<NBeats>(*ctip, _nbeats_stacks);
+            // load params if any after instanciation
+            if (!this->_mlmodel._native.empty())
+              torch::load(_module._native, this->_mlmodel._native, _device);
+            _module._native->to(_device);
+          }
+        if (_module._graph)
```

Comment:
`else` applicable ?

Reply:
same as before, one could want to chain a native and a graph, but with current code they take same input